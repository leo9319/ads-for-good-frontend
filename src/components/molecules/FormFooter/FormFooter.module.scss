@import '../../../styles/colours';

.formContainer {
  display: grid;
  gap: 4px;

  .formFooterQuest {
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 4px;
  }

  .description {
    display: flex;
    justify-content: center;
    text-align: center;
    gap: 4px;

    .text {
      color: $color-neutral-600;
      font-size: 14px;
      line-height: 16px;
    }
  }
}
