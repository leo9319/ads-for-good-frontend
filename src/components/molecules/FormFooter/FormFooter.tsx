import React, { ReactElement } from 'react';
import Flex from '@radix-styles/atoms/Flex';
import Text from '@radix-styles/atoms/Text';
import Link from '@radix-styles/atoms/Link';
import styles from './FormFooter.module.scss';

/**
 * @interface
 * The FormFooter props type of {@link FormFooter}.
 */
export interface FormFooterProps {
  /**
   * Content/question that will appear before the text link.
   * @property
   */
  content: string;
  /**
   * Text for the Hyperlink
   * @property
   */
  linkText: string;
  /**
   * Text content to display the description
   * @property
   */
  description?: string;
  /**
   * Hyperlink for the navigation
   * @property
   */
  href?: string;
  /**
   * Callback function triggered when the login link is clicked
  @property
  */
  linkClick?: () => void;
}

/**
 * FormFooter component
 *
 * @group Molecules
 * @category Component
 *
 * @param {FormFooterProps} props - The FormFooter props
 * @returns {ReactElement} - FormFooter Component
 */
export const FormFooter = ({
  content,
  linkText,
  description,
  href,
  linkClick,
}: FormFooterProps): ReactElement => {
  return (
    <Flex className={styles.formContainer}>
      <Flex className={styles.formFooterQuest}>
        <Text>{content}</Text>
        <Link onClick={linkClick} targetHref={href}>
          {linkText}
        </Link>
      </Flex>
      {description && (
        <Flex className={styles.description}>
          <Text className={styles.text}>{description}</Text>
        </Flex>
      )}
    </Flex>
  );
};

export default FormFooter;
