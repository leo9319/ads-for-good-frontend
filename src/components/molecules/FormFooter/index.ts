export type { FormFooterProps } from './FormFooter';
export { default } from './FormFooter';
export { FormFooter } from './FormFooter';
