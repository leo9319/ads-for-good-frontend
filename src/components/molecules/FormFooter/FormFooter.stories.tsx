import FormFooter from './FormFooter';
import { StoryObj, Meta } from '@storybook/react';

const meta = {
  title: 'Library/Molecules/FormFooter',
  component: FormFooter,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
  },
  argTypes: {
    content: {
      control: { type: 'text' },
    },
    linkText: {
      control: { type: 'text' },
    },
    description: {
      control: { type: 'text' },
    },
    href: {
      control: { type: 'text' },
    },
  },
} satisfies Meta<typeof FormFooter>;

export default meta;

type Story = StoryObj<typeof meta>;

const description = 'With your help, we can end povertyâ€”for good.';
export const SignUpWithTextLink: Story = {
  args: {
    content: "Don't have an account?",
    linkText: 'Sign Up',
    description: description,
  },
};
export const SignUpWithHyperLink: Story = {
  args: {
    content: "Don't have an account?",
    linkText: 'Sign Up',
    description: description,
    href: 'https://dev-aem.worldvision.ca/en/register.html',
  },
};
