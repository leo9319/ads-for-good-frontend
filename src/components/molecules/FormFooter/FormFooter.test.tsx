import React from 'react';
import { render } from '@testing-library/react';
import FormFooter from './FormFooter';

describe('FormFooter', () => {
  it('Should render successfully', () => {
    const { baseElement } = render(
      <FormFooter content="Don't have an account?" linkText="Sign Up" />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
