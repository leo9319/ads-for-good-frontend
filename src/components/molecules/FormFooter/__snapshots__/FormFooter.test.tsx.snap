// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFooter Should render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-Flex formContainer"
    >
      <div
        class="rt-Flex formFooterQuest"
      >
        <span
          class="rt-Text typography-font-size-3 typography-font-weight-regular typography-line-height-3 typography-font-family-default"
        >
          Don't have an account?
        </span>
        <a
          class="rt-Text rt-reset rt-Link rt-underline-auto link body-md"
          data-accent-color=""
          tabindex="0"
        >
          Sign Up
        </a>
      </div>
    </div>
  </div>
</body>
`;
